pkg/admission/imagepolicy/accept.go

```diff
@@ -58,13 +58,22 @@ func accept(accepter rules.Accepter, policy imageResolutionPolicy, resolver imag
 					decision.resolutionErr = err
 
 				case err == nil:
+					oldDecissionAttributes := decision.attrs
 					// if we resolved properly, assign the attributes and rewrite the pull spec if we need to
 					decision.attrs = resolvedAttrs
 
 					if policy.RewriteImagePullSpec(resolvedAttrs, attr.GetOperation() == admission.Update, gr) {
-						ref.Namespace = ""
```

Comment:
I know the tests are failing here. What are the other use cases that failing here with this image is prohibited by policy (changed after admission) makes sense?